From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ronald Claveau <linux-kernel-dev@aliel.fr>
Date: Fri, 3 Apr 2026 12:08:16 +0200
Subject: dt-bindings: phy: amlogic: usb2: Add compatible for Amlogic T7 SoC

The Amlogic T7 USB2 PHY uses the same register layout
and initialization sequence as the G12A USB2 PHY.
Add amlogic,t7-usb2-phy with a fallback to amlogic,g12a-usb2-phy.

Signed-off-by: Ronald Claveau <linux-kernel-dev@aliel.fr>
---
 Documentation/devicetree/bindings/phy/amlogic,g12a-usb2-phy.yaml | 13 +++++++---
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/Documentation/devicetree/bindings/phy/amlogic,g12a-usb2-phy.yaml b/Documentation/devicetree/bindings/phy/amlogic,g12a-usb2-phy.yaml
index 111111111111..222222222222 100644
--- a/Documentation/devicetree/bindings/phy/amlogic,g12a-usb2-phy.yaml
+++ b/Documentation/devicetree/bindings/phy/amlogic,g12a-usb2-phy.yaml
@@ -12,10 +12,15 @@ maintainers:
 
 properties:
   compatible:
-    enum:
-      - amlogic,g12a-usb2-phy
-      - amlogic,a1-usb2-phy
-      - amlogic,s4-usb2-phy
+    oneOf:
+      - items:
+          - enum:
+              - amlogic,t7-usb2-phy
+          - const: amlogic,g12a-usb2-phy
+      - enum:
+          - amlogic,g12a-usb2-phy
+          - amlogic,a1-usb2-phy
+          - amlogic,s4-usb2-phy
 
   reg:
     maxItems: 1
-- 
Armbian

